page/effects/queue-and-dequeue-explained.md

```diff
@@ -4,53 +4,135 @@
 	"source": "http://jqueryfordesigners.com/api-queue-dequeue/"
 }</script>
 
-When you use the `.animate()`, `.show()`, `.hide()`, `.slideUp()`, etc. effect methods, you're adding a job to the effects queue. By default, using `.queue()` and passing a function, will add it to the effects queue. So we're creating our own bespoke animation step:
+Queues are the foundation for all animations in jQuery, they allow a series functions to be executed asynchronously on an element.  Methods such as `.slideUp()`, `.slideDown()`, `.fadeIn()`, and `.fadeOut()` all use `.animate()`, which leverages *queues* to build up the series of steps that will transition one or more CSS values throughout the duration of the animation.
+
+We can pass a callback function to the `.animate()` method, which will execute once the animation has completed.
 
 ```
 $( ".box" )
-	.animate({
+	.animate( {
 		height: 20
-	}, "slow" )
-	.queue(function() {
+	}, "slow", function() {
 		$( "#title" ).html( "We're in the animation, baby!" );
-	});
+	} );
 ```
 
-As I said though, these methods come in pairs, so anything you add using `.queue()`, you need to dequeue to allow the process to continue. In the code above, if I chained more animations on, until I call `$( this ).dequeue()`, the subsequent animations wouldn't run:
+## Queues As Callbacks
+
+Instead of passing a callback as an argument, we can add another function to the *queue* that will act as our callback. This will execute after all of the steps in the animation have completed.
 
 ```
 $( ".box" )
-	.animate({
+	.animate( {
 		height: 20
-	}, "slow" )
-	.queue(function() {
+	}, "slow")
+	.queue( function() {
 		$( "#title" ).html( "We're in the animation, baby!" );
-		$( this ).dequeue();
-	}).animate({
-		height: 150
-	});
+	} );
+
 ```
 
-Keeping in mind that the animation won't continue until we've explicitly called `.dequeue()`, we can easily create a pausing plugin, by adding a step in the queue that sets a timer and triggers after `delay` milliseconds, at which time, it dequeues the element:
+To add multiple functions to the queue, you can call `.queue()` multiple times.
 
 ```
-$.fn.pause = function( delay ) {
-	return this.queue(function() {
-		var elem = this;
-		setTimeout(function() {
-			return $( elem ).dequeue();
-		}, delay );
-	});
-};
-
 $( ".box" )
-	.animate({
+	.animate( {
 		height: 20
 	}, "slow" )
-	.pause( 1000 )
-	.animate({
-		height: 150
-	});
+	.queue( function() {
+		console.log('I fired!');
+	} )
+	.animate( {
+		height: 50
+	}, "fast" )
+	.queue( function() {
+		console.log('I fired too!');
```

Comment:
> only it didn't... Why show the wrong way?

Same thing as previous comment, but you're right.

> I say you should add the next method to all these examples (instead of dequeue, I find it cleaner)

Yup